README.md

```diff
@@ -1,2 +1,422 @@
+
+✅ Simple integration: no DI-specific types or generics required
+
+✅ Easy testing: every type has a memberwise initializer
+
+✅ Clear error messages: never debug generated code
+
+## Using SafeDI
+
+SafeDI utilizes Swift Macros and Swift Package Manager plugins to read your code and generate a dependency tree that is validated at compile time. Dependencies can either be instantiated by SafeDI or forwarded into the SafeDI dependency tree.
+
+Opting a type into the SafeDI dependency tree is straightforward: add the `@Instantiable` macro to your type declaration, and decorate your type‘s dependencies with macros that signal the lifecycle of each property. Decorate a property with `@Instantiated` if you want to initialize it when the enclosing type is initialized; `@Forwarded` if you want to propagate a runtime-determine value down the dependency tree; or `@Received` if you want to receive the property from an `@Instantiated` or `@Forwarded` property further up the dependency tree.
+
+If a type is declared in third-party code, you can declare an extension the type in your code and decorate it with the `@ExternalInstantiable` macro to opt it into SafeDI.
+
+Let‘s walk through each of these macros in detail.
+
+### @Instantiable
+
+Type declarations decorated with the [`@Instantiable` macro](Sources/SafeDI/PropertyDecoration/Instantiable.swift) are able to be instantiated by SafeDI. Types decorated with this macro can instantiate other `@Instantiable` dependencies, forward dependencies injected from outside of SafeDI, or receive dependencies instantiated or forwarded by objects further up the dependency tree.
+
+Every `@Instantiable`-decorated type must be:
+
+1. `public` or `open`
+
+2. Have a `public init(…)` or `open init(…)` method that receives every injectable property
+
+The `@Instantiable` guides engineers through satisfying these requirements with build-time FixIts.
+
+#### Example
+
+Here is a sample `UserService` implementation that is `@Instantiable`:
+
+```swift
+import SafeDI
+
+/// A protocol that defines a UserService.
+/// It is not necessary to utilize protocols with SafeDI, but since
+/// protocol-driven development aids both testability and dependency
+/// inversion, our examples show how protocols can be used with SafeDI.
+public protocol UserService {
+    var user: User? { get }
+    func login(username: String, password: String) async throws -> User
+}
+
+/// A default implementation of `UserService` that can fulfill `@Instantiated`
+/// properties of type `UserService` or `DefaultUserService`.
+@Instantiable(fulfillingAdditionalTypes: [UserService.self])
+public final class DefaultUserService: UserService {
+
+    // MARK: Initialization
+
+    /// Public, memberwise initializer that takes each injected property.
+    public init(authService: AuthService, securePersistentStorage: SecurePersistentStorage) {
+        self.authService = authService
+        self.securePersistentStorage = securePersistentStorage
+    }
```

Comment:
Any chance this initializer too can be generated?

Reply:
In the simple case, it can! I've got [a TODO in the codebase to simplify this](https://github.com/dfed/SafeDI/blob/76b713a766232f203f72e1eeb303d01a81d69cfc/Sources/SafeDIMacros/Macros/InstantiableMacro.swift#L83).

When you have non-injected properties that must be set in `init`, though, you'll still need to write your own initializer. I like that today we can generate the structure of the initializer for you with a FixIt – if I started generating the initializer for you we'd lose the FixIt. I haven't figured out if the trade-off of simplifying the simple case while making the more complex (but still common!) case a bit more annoying is worth it yet.

Of course, now that I've typed that out... I could generate the FixIt only if you have a local property with an initializer... that'd be the best of both worlds. We’ll Get There™. Just gotta put in the time 🙂